서영/-2진수.py

```diff
@@ -0,0 +1,20 @@
+# https://www.acmicpc.net/problem/2089
+# 2진수: 2로 나눈 나머지로 / -2진수: -2로 나눈 나머지로?
+# -13 -> 7(1) -> -3(1) -> 2(1) -> -1(0) -> 1(1)
+
+from sys import stdin
+
+N = int(stdin.readline())
+
+ans = ""
+while N > 1 or N < 0:
+    if N % 2:
+        ans += "1"
+        N = N // -2 + 1
+    else:
+        ans += "0"
+        N = N // -2
+
+ans += "1" if N else "0"
```

Comment:
아래의 그림과 같이(-2진수가 아니지만) ➡️⬆️⬆️ 이런 식으로 구하고 계신 것으로 생각이 됩니다!
![image](https://user-images.githubusercontent.com/26241585/92204067-9e487600-eebd-11ea-9fd1-547df30a5de3.png)

하지만 아래의 그림과 같이(-2진수가 아니지만) ⬆️⬆️⬆️ 이런 식으로 구하게 된다면
![image](https://user-images.githubusercontent.com/26241585/92204151-d059d800-eebd-11ea-8eb0-3ad0427544d8.png)

대략 아래의 코드와 같이 줄일 수 있지 않을까 생각됩니다!

```suggestion
while N:
    if N % 2:
        ans += "1"
        N = N // -2 + 1
    else:
        ans += "0"
        N = N // -2
```

Reply:
오,, 그렇네요👍